internal/storagetools/get_object_handler.go

```diff
@@ -0,0 +1,105 @@
+package storagetools
+
+import (
+	"errors"
+	"io"
+	"os"
+	"path"
+
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/compression"
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type DownloadMode string
+
+const (
+	DownloadRaw        DownloadMode = "raw"
+	DownloadDecompress DownloadMode = "decompress"
+	DownloadDecrypt    DownloadMode = "decrypt"
+)
+
+func HandleGetObject(objectPath, dstPath string, folder storage.Folder, mode DownloadMode) {
+	fileName := path.Base(objectPath)
+	targetPath, err := getTargetFilePath(dstPath, fileName)
+	tracelog.ErrorLogger.FatalfOnError("Failed to determine the destination path: %v", err)
+
+	dstFile, err := os.OpenFile(targetPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC|os.O_EXCL, 0666)
```

Comment:
0640 is more safe mask.    backup files may contain sensetive data.

Reply:
Fixed in 947351b